packages/client/src/Call.ts

```diff
@@ -342,16 +342,18 @@ export class Call {
     );
 
     this.leaveCallHooks.add(
-      // watch for auto drop cancellation
-      createSubscription(this.state.callingState$, (callingState) => {
+      // cancel auto-drop when call is
+      createSubscription(this.state.session$, (session) => {
         if (!this.ringing) return;
-        if (
-          callingState === CallingState.JOINED ||
-          callingState === CallingState.JOINING ||
-          callingState === CallingState.LEFT
-        ) {
```

Comment:
Cancelling auto-drop on calling state change is no longer required, since it's covered by two other checks:
1. Auto-drop is cancelled if the call is either accepted or rejected by the current user.
2. Auto-drop is a no-op if calling state changed to anything other than `RINGING`.